### DISCUSSION TITLE
Refactoring of the ExtensionAnnotationProcessor

### DISCUSSION THREAD

**Discussion by renanmachad** [quarkusio/quarkus#41704] [Design Discussions]
Hello, in the ExtensionAnnotationProcessor class we have this code, there are many try-with-resources instructions, but this can be simplified,
is it justified to open a PR to improve this?
I'm new to the open source community and I'm looking for ways to help the framework I use daily.
Thanks!
`private void writeListResourceFile(Collection<String> crListClasses, FileObject listResource) throws IOException {
        try (OutputStream os = listResource.openOutputStream()) {
            try (BufferedOutputStream bos = new BufferedOutputStream(os)) {
                try (OutputStreamWriter osw = new OutputStreamWriter(bos, StandardCharsets.UTF_8)) {
                    try (BufferedWriter bw = new BufferedWriter(osw)) {
                        for (string item: crListClasses) {
                            bw.write(item);
                            bw.newLine();
                        }
                    }
                }
            }
        }
    }`

**Comment by cescoffier** [comment]
Sure! Please open a PR; it would be easier to understand how it can be simplified.

**Comment by gsmet** [comment]
Note that all this code has been written before we switched to Java 17 so don't hesitate to use new constructs to simplify this stuff.

**Comment by renanmachad** [comment]
Okay, very thanks!
